src/app/components/POCs/ProcessorEditHeader/ProcessorEditHeader.stories.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useState } from "react";
+import { ComponentStory, ComponentMeta } from "@storybook/react";
+import ProcessorEditHeader from "@app/components/POCs/ProcessorEditHeader/ProcessorEditHeader";
+
+export default {
+  title: "PoCs/Processor Edit Header",
+  component: ProcessorEditHeader,
+  args: {
+    onCreate: () => {},
+    showCreateAction: true,
+  },
+} as ComponentMeta<typeof ProcessorEditHeader>;
+
+const Template: ComponentStory<typeof ProcessorEditHeader> = (args) => {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { onNameChange, name, ...rest } = args;
+  const [value, setValue] = useState(args.name);
+
+  return (
+    <div style={{ padding: "3em 0" }}>
+      <ProcessorEditHeader
+        onNameChange={(newValue): void => setValue(newValue)}
```

Comment:
@kelvah Please what is the purpose of this code? Does it change the 'name' attribute when I set it in story book UI?

Reply:
Hi @sjamboro, it's meant to demonstrate the component behavior when trying to set a new name.

`ProcessorEditHeader` keeps an internal state for the name value only to manage the editing phase, meaning controlling the input field and allowing canceling changes.

After renaming the processor and confirming the changes by clicking on the tick icon, the new value is passed up to its parent via `onNameChange`, where the actual state for the name value is stored.

That's the reason why some state management is needed inside the story, otherwise confirming the changes would have no visible effect inside storybook.